base/applications/mspaint/selectionmodel.cpp

```diff
@@ -207,6 +207,7 @@ void SelectionModel::InsertFromHBITMAP(HBITMAP hBm, INT x, INT y)
     m_rc.top = y;
     m_rc.right = m_rc.left + GetDIBWidth(hBm);
     m_rc.bottom = m_rc.top + GetDIBHeight(hBm);
+    ::SetRect(&m_rcOld, -2, -2, -1, -1); // Outside of image
```

Comment:
Which problem does this fix?

Reply:
Pasting bug.

When Rectangular Selection, if selection area was not moved, then it don't update the undo buffer.